sql/structured.go

```diff
@@ -178,11 +183,13 @@ func (desc *TableDescriptor) AllocateIDs() error {
 		}
 	}
 
-	// This is sort of ugly. We want to make sure the descriptor is valid, except
-	// for checking the table ID. So we whack in a valid table ID for the
-	// duration of the call to Validate.
+	// This is sort of ugly. If the descriptor does not have an ID, we hack one in
```

Comment:
I really don't like this. I'll see if I can change it to return an error if the input is bad, rather than call validate at the end.
Validate would then be called by a "write descriptor" helper (not the existing one, that one should really be createDescriptor). This would ensure that validate is always called on the full descriptor at write time.